### PR TITLE
GO CLI Lib Add default Templates. Refactor Execute and define Interface Verified

### PR DESCRIPTION
**Pull request by tejal29**
Depends on PR  #551  and #533 

Review Commit 
https://github.com/GoogleCloudPlatform/runtimes-common/pull/556/commits/814fea1f4d5cc3f25eccd2b9e399591a5f1850a1

### CODE REVIEW

File: ctc_lib/CLI_interface.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 Google Inc. All Rights Reserved.
```
Comment (dlorenc) [inline ctc_lib/CLI_interface.go]: Could you name this all lowercase?

Comment (dlorenc) [reply ctc_lib/CLI_interface.go]: The file, not the structs.

File: ctc_lib/CLI_interface.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	ValidateCommand() error
+	IsRunODefined() bool
+	Init()
+}
+
+func Execute(ctb CLIInterface) (err error) {
+	defer errRecover(&err)
+
+	if err := ctb.ValidateCommand(); err != nil {
+		CommandExit(err)
```
Comment (dlorenc) [inline ctc_lib/CLI_interface.go]: This function can return an error, should we do that here instead of exiting?

File: ctc_lib/CLI_interface.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			ctb.PrintO(c, args)
+		}
+		ctb.SetRun(cobraRun)
+	}
+
+	err = ctb.GetCommand().Execute()
+
+	//Add empty line as template.Execute does not print an empty line
+	ctb.GetCommand().Println()
+	CommandExit(err)
```
Comment (dlorenc) [inline ctc_lib/CLI_interface.go]: It's a little weird to collect the error, then call exit, then return it. Can CommandExit return the error of it doesn't exit for some reason? Then we can return CommandExit

File: ctc_lib/CLI_interface.go (hunk 1 above)
Comment (tejal29) [reply ctc_lib/CLI_interface.go]: done

File: ctc_lib/CLI_interface.go (hunk 3 above)
Comment (tejal29) [reply ctc_lib/CLI_interface.go]: yeah, i think i wanted to define ExecuteE which returns Error and Execute() which does os.Exit.
  Forgot to make those changes.

Comment (tejal29) [reply ctc_lib/CLI_interface.go]: done

**Pull request merged**